src/routes/_surfaces/workspace/explore/time-range-constructor/timeRangeUtils.ts

```diff
@@ -0,0 +1,589 @@
+        year: "numeric",
+        month: "short",
+        day: "numeric",
+      });
+    case "7 day":
+      return new Date(date).toLocaleDateString(undefined, {
+        year: "numeric",
+        month: "short",
+        day: "numeric",
+      });
+    case "1 month":
+      return new Date(date).toLocaleDateString(undefined, {
+        year: "numeric",
+        month: "short",
+      });
+    case "1 year":
+      return new Date(date).toLocaleDateString(undefined, {
+        year: "numeric",
+      });
+    default:
+      throw new Error(`Unknown interval: ${interval}`);
+  }
+};
+
+export const prettyTimeGrain = (timeGrain: TimeGrain): string => {
+  if (!timeGrain) return "";
+  switch (timeGrain) {
+    // case TimeGrain.FiveMinutes:
+    //   return "5 minute";
+    // case TimeGrain.FifteenMinutes:
+    //   return "15 minute";
+    case TimeGrain.OneHour:
+      return "hourly";
+    case TimeGrain.OneDay:
+      return "daily";
+    // case TimeGrain.OneWeek:
+    //   return "weekly";
+    case TimeGrain.OneMonth:
+      return "monthly";
+    case TimeGrain.OneYear:
+      return "yearly";
+    default:
+      throw new Error(`Unknown time grain: ${timeGrain}`);
+  }
+};
+
+const getTimeRangeDuration = (
+  timeRangeName: TimeRangeName,
+  allTimeRange: TimeSeriesTimeRange
+): number => {
+  switch (timeRangeName) {
+    // case TimeRangeName.LastHour:
+    //   return 60 * 60 * 1000;
+    case TimeRangeName.Last6Hours:
+      return 6 * 60 * 60 * 1000;
+    case TimeRangeName.LastDay:
+      return 24 * 60 * 60 * 1000;
+    case TimeRangeName.Last2Days:
+      return 2 * 24 * 60 * 60 * 1000;
+    case TimeRangeName.Last5Days:
+      return 5 * 24 * 60 * 60 * 1000;
+    case TimeRangeName.LastWeek:
+      return 7 * 24 * 60 * 60 * 1000;
+    case TimeRangeName.Last2Weeks:
+      return 2 * 7 * 24 * 60 * 60 * 1000;
+    case TimeRangeName.Last30Days:
+      return 30 * 24 * 60 * 60 * 1000;
+    case TimeRangeName.Last60Days:
+      return 60 * 24 * 60 * 60 * 1000;
+    case TimeRangeName.AllTime:
+      return (
+        new Date(allTimeRange.end).getTime() -
+        new Date(allTimeRange.start).getTime()
+      );
+    default:
+      throw new Error(`Unknown time range name: ${timeRangeName}`);
+  }
+};
+
+const getTimeGrainDuration = (timeGrain: TimeGrain): number => {
+  switch (timeGrain) {
+    // case TimeGrain.FiveMinutes:
+    //   return 5 * 60 * 1000;
+    // case TimeGrain.FifteenMinutes:
+    //   return 15 * 60 * 1000;
+    case TimeGrain.OneHour:
+      return 60 * 60 * 1000;
+    case TimeGrain.OneDay:
+      return 24 * 60 * 60 * 1000;
+    // case TimeGrain.OneWeek:
+    //   return 7 * 24 * 60 * 60 * 1000;
+    case TimeGrain.OneMonth:
+      return 30 * 24 * 60 * 60 * 1000;
+    case TimeGrain.OneYear:
+      return 365 * 24 * 60 * 60 * 1000;
+    default:
+      throw new Error(`Unknown time grain: ${timeGrain}`);
+  }
+};
+
+// const roundDateDown = (date: Date | undefined, timeGrain: TimeGrain): Date => {
```

Comment:
could you remove this commented-out code?

Reply:
Yep, just did. It's used in weekly time grains, which I've temporarily disabled, but I'll remove the code for now.